streams/src/test/java/org/apache/kafka/streams/tests/SmokeTestDriver.java

```diff
@@ -357,256 +274,197 @@ public static void verify(final String kafka, final Map<String, Set<Integer>> al
         }
 
         boolean success;
-        success = allData.equals(received);
+
+        final Map<String, HashSet<Number>> received =
+            events.get("echo")
+                  .entrySet()
+                  .stream()
+                  .map(entry -> mkEntry(entry.getKey(), new HashSet<>(entry.getValue())))
+                  .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+        success = inputs.equals(received);
 
         if (success) {
             System.out.println("ALL-RECORDS-DELIVERED");
         } else {
             int missedCount = 0;
-            for (final Map.Entry<String, Set<Integer>> entry : allData.entrySet()) {
+            for (final Map.Entry<String, Set<Integer>> entry : inputs.entrySet()) {
                 missedCount += received.get(entry.getKey()).size();
             }
             System.out.println("missedRecords=" + missedCount);
         }
 
-        success &= verifyMin(min, allData, true);
-        success &= verifyMax(max, allData, true);
-        success &= verifyDif(dif, allData, true);
-        success &= verifySum(sum, allData, true);
-        success &= verifyCnt(cnt, allData, true);
-        success &= verifyAvg(avg, allData, true);
-        success &= verifyTAgg(tagg, allData, true);
+        // give it one more try if it's not already passing.
+        if (!verificationResult.passed()) {
+            verificationResult = verifyAll(inputs, events);
+        }
+        success &= verificationResult.passed();
+
+        System.out.println(verificationResult.result());
 
         System.out.println(success ? "SUCCESS" : "FAILURE");
+        return verificationResult;
     }
 
-    private static boolean verifyMin(final Map<String, Integer> map, final Map<String, Set<Integer>> allData, final boolean print) {
-        if (map.isEmpty()) {
-            if (print) {
-                System.out.println("min is empty");
+    private static Number deserializeValue(final ConsumerRecord<String, byte[]> record) {
+        final Number value;
+        switch (record.topic()) {
+            case "data": {
+                value = intSerde.deserializer().deserialize("", record.value());
+                break;
             }
-            return false;
-        } else {
-            if (print) {
-                System.out.println("verifying min");
+            case "echo": {
+                value = intSerde.deserializer().deserialize("", record.value());
```

Comment:
This is s repetition of `"data"` case -- similar below -- we should put all int/long/double cases etc together to void code duplication using "case-fall-through" pattern.

Reply:
I actually did that in my follow-up PR. I'll go ahead and pull it into this one.